packages/mds-transactions-service/repository/index.ts

```diff
@@ -0,0 +1,177 @@
+import { InsertReturning, RepositoryError, ReadWriteRepository } from '@mds-core/mds-repository'
+import { NotFoundError } from '@mds-core/mds-utils'
+import { UUID } from '@mds-core/mds-types'
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import { LessThan, MoreThan, Between } from 'typeorm'
+import {
+  TransactionDomainModel,
+  TransactionOperationDomainModel,
+  TransactionSearchParams,
+  TransactionStatusDomainModel
+} from '../@types'
+import {
+  TransactionEntityToDomain,
+  TransactionDomainToEntityCreate,
+  TransactionOperationEntityToDomain,
+  TransactionOperationDomainToEntityCreate,
+  TransactionStatusEntityToDomain,
+  TransactionStatusDomainToEntityCreate
+} from './mappers'
+import { TransactionEntity } from './entities/transaction-entity'
+import { TransactionOperationEntity } from './entities/operation-entity'
+import { TransactionStatusEntity } from './entities/status-entity'
+import migrations from './migrations'
+
+class TransactionReadWriteRepository extends ReadWriteRepository {
+  public getTransaction = async (transaction_id: UUID): Promise<TransactionDomainModel> => {
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      const entity = await connection.getRepository(TransactionEntity).findOne({
+        where: {
+          transaction_id
+        }
+      })
+      if (!entity) {
+        throw new NotFoundError(`Transaction ${transaction_id} not found`)
+      }
+      return TransactionEntityToDomain.map(entity)
+    } catch (error) {
+      throw RepositoryError(error)
+    }
+  }
+
+  // TODO search criteria, paging
+  public getTransactions = async (search: TransactionSearchParams): Promise<TransactionDomainModel[]> => {
+    const { connect } = this
+    try {
+      const connection = await connect('ro')
+      const where: any = {}
+      if (search.provider_id) {
+        where.provider_id = search.provider_id
+      }
+      if (search.start_timestamp && search.end_timestamp) {
+        where.timestamp = Between(search.start_timestamp, search.end_timestamp)
+      } else if (search.start_timestamp) {
+        where.timestamp = MoreThan(search.start_timestamp)
+      } else if (search.end_timestamp) {
+        where.timestamp = LessThan(search.end_timestamp)
+      }
```

Comment:
Self-executing closures & functional programming FTW!

```suggestion
    const where = (() => {
      const { provider_id, start_timestamp, end_timestamp } = search

      const timestamp = (() => {
          if (start_timestamp && end_timestamp) return Between(start_timestamp, end_timestamp)
          if (start_timestamp) return MoreThan(start_timestamp)
          if (end_timestamp) return LessThan(end_timestamp)
      })()

      return { provider_id, timestamp }
})()
```

Reply:
eh